pkg/services/ngalert/notifier/channels/channels_validation.go

```diff
@@ -0,0 +1,215 @@
+package channels
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+)
+
+func ValidateContactPointReceiverWithSecure(channelType string, settings *simplejson.Json, secureSettings map[string][]byte, fn GetDecryptedValueFn) (bool, error) {
+	// create a flat map of settings
+	copy := *settings
+	for key := range secureSettings {
+		copy.Set(key, fn(context.Background(), secureSettings, key, settings.Get(key).MustString()))
+	}
+	return ValidateContactPointReceiver(channelType, &copy)
+}
+
+//nolint:gocyclo
+func ValidateContactPointReceiver(channelType string, settings *simplejson.Json) (bool, error) {
+	switch strings.ToLower(channelType) {
```

Comment:
Usually, (at least in OO languages, which this is not), having a large switch statement that switches on a bunch of different types is a code smell - specifically, this goes against the Open-Closed principle. Usually I see devs preferring to move in the opposite direction and break up a large switch across types into each individual sub-type.

I'm a little concerned about maintainability here, both because it's easy to forget to add new cases, and it's also easy for the code in each case to get out of sync. This function could grow unbounded-ly in size over time as we add new options and channels. Is there clean a way to break this up into a per-channel basis?